### PR TITLE
Function for plotting shear force added

### PR DESCRIPTION
**Pull request by Psycho-Pirate**
#### References to other Issues or PRs

See #19202

#### Brief description of what is fixed or changed
A function for plotting shear force in 3D Beam is added.

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by Psycho-Pirate** [comment]
Here is an example:
```
>>> from sympy.physics.continuum_mechanics.beam import Beam3D
>>> from sympy import symbols
>>> l, E, G, I, A, x = symbols('l, E, G, I, A, x')
>>> b = Beam3D(30, E, G, I, A, x)
>>> b.apply_load(8, start=0, order=0, dir="y")
>>> b.apply_load(9*x, start=0, order=0, dir="z")
>>> b.bc_deflection = [(0, [0, 0, 0]), (30, [0, 0, 0])]
>>> R1, R2, R3, R4 = symbols('R1, R2, R3, R4')
>>> b.apply_load(R1, start=0, order=-1, dir="y")
>>> b.apply_load(R2, start=30, order=-1, dir="y")
>>> b.apply_load(R3, start=0, order=-1, dir="z")
>>> b.apply_load(R4, start=30, order=-1, dir="z")
>>> b.solve_for_reaction_loads(R1, R2, R3, R4)
>>> print(b.plot_shear_force())
```
image
image
image

**Comment by Psycho-Pirate** [comment]
Ping @moorepants @ishanaj @Sc0rpi0n101

**Comment by Psycho-Pirate** [comment]
Please add the physics.continuum.mechanics and plotting labels

**Comment by Psycho-Pirate** [comment]
I have added a function for plotting shear force.
If this approach is approved, I will add other functions as well.
@moorepants @ishanaj @jashan498

**Comment by czgdp1807** [comment]
@moorepants Is this good to go? Any thoughts on the changes by @Psycho-Pirate ?

File: sympy/physics/continuum_mechanics/beam.py
Comment (ishanaj) [inline sympy/physics/continuum_mechanics/beam.py:2309]: This case should plot all the graphs as subplots in a single plot.

**Comment by ishanaj** [comment]
I think for all the different cases the same code is being repeated. It would be better if the code is separated and put inside a helper function which would return the plot whenever required

File: sympy/physics/continuum_mechanics/beam.py
Comment (Psycho-Pirate) [reply sympy/physics/continuum_mechanics/beam.py:2309]: The values of shear force can vary greatly along different directions. See the example i mentioned above, shear force along y has smaller values in hundreds, while shear force along z has values in thousands. Having them on a same plot may not convey the information clearly. What do you suggest?

**Comment by Psycho-Pirate** [comment]
> I think for all the different cases the same code is being repeated. It would be better if the code is separated and put inside a helper function which would return the plot whenever required

I agree! I will start working on this.

File: sympy/physics/continuum_mechanics/beam.py
Comment (ishanaj) [reply sympy/physics/continuum_mechanics/beam.py:2309]: > The values of shear force can vary greatly along different directions

I don't think that would be an issue. Just give it a try and share the plot results here. You can see how `plot_loading_results()` functions.

**Comment by Psycho-Pirate** [comment]
Please have a look @ishanaj

**Comment by czgdp1807** [comment]
@ishanaj Please review this PR.

File: sympy/physics/continuum_mechanics/beam.py
Comment (ishanaj) [inline sympy/physics/continuum_mechanics/beam.py:2238]: It's better to name a helper function with an underscore, like: `_plot_shear_force()`.

File: sympy/physics/continuum_mechanics/beam.py
Comment (ishanaj) [inline sympy/physics/continuum_mechanics/beam.py:2281]: Also, you should possibly add a basic example in the documentation so that the user knows how to use this function.

**Comment by ishanaj** [comment]
It would be great if you could share some screenshots for the fourth case (`dir='all' `) too.
The rest looks good.

File: sympy/physics/continuum_mechanics/beam.py
Comment (ishanaj) [inline sympy/physics/continuum_mechanics/beam.py:2263]: Better to use different `line_color` for different axis.

Comment (ishanaj) [reply sympy/physics/continuum_mechanics/beam.py:2263]: Also, you can either directly return `plot(......)` function or choose a better variable name rather than `P`

**Comment by Psycho-Pirate** [comment]
> It would be great if you could share some screenshots for the fourth case (`dir='all' `) too.
> The rest looks good.

I have made the suggested changes and added an example in the docs as well.
```
>>> from sympy.physics.continuum_mechanics.beam import Beam3D
>>> from sympy import symbols
>>> l, E, G, I, A, x = symbols('l, E, G, I, A, x')
>>> b = Beam3D(20, E, G, I, A, x)
>>> b.apply_load(15, start=0, order=0, dir="z")
>>> b.apply_load(12*x, start=0, order=0, dir="y")
>>> b.bc_deflection = [(0, [0, 0, 0]), (20, [0, 0, 0])]
>>> R1, R2, R3, R4 = symbols('R1, R2, R3, R4')
>>> b.apply_load(R1, start=0, order=-1, dir="z")
>>> b.apply_load(R2, start=20, order=-1, dir="z")
>>> b.apply_load(R3, start=0, order=-1, dir="y")
>>> b.apply_load(R4, start=20, order=-1, dir="y")
>>> b.solve_for_reaction_loads(R1, R2, R3, R4)
>>> b.plot_shear_force('y')
```
Figure_1
```
>>> b.plot_shear_force()  #for all directions
```
xyz_1

**Comment by Psycho-Pirate** [comment]
Here's a screenshot of the docs with the example added.
Screenshot from 2020-12-26 16-30-50

**Comment by ishanaj** [comment]
Great work! Looks good to me. 
@Psycho-Pirate are you intending to add other plotting functions too in this PR?

**Comment by Psycho-Pirate** [comment]
> Great work! Looks good to me.
> @Psycho-Pirate are you intending to add other plotting functions too in this PR?

Thanks!
I'm not planning about adding more plotting functions as of now because i have other issues pending and this pr has been open for months. But I'm open to suggestions about new plotting functions if you have any.

**Comment by ishanaj** [comment]
A plot function for bending moment, deflection, slope, should also be built in a similar manner. Maybe they can be added in a separate PR. Otherwise, I think this is good to go! @moorepants @czgdp1807

**Comment by Psycho-Pirate** [comment]
> A plot function for bending moment, deflection, slope, should also be built in a similar manner.

I'll start working on them. Thanks for reviewing!

**Comment by Psycho-Pirate** [comment]
Please get this merged @czgdp1807 @moorepants

**Comment by czgdp1807** [comment]
@asmeurer Could you please give merge access to @ishanaj ?
Many days have passed since merge approval was given and therefore I will wait till tomorrow to merge this. Thanks for the contributions.

**Comment by oscarbenjamin** [comment]
I have sent a request to add @ishanaj as a member. @ishanaj if you accept the request then you will be able to merge this.

**Comment by ishanaj** [comment]
Thanks @oscarbenjamin! I will be merging this now.

**Review by ishanaj — APPROVED** [review]

**Pull request merged**
